packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list/page-menu.tsx

```diff
@@ -107,12 +111,24 @@ export const PageMenu: FunctionComponent<PageMenuProps> = ({
     ],
     [copied, popupState, createSubPage, accountId, entityId, archivePage],
   );
+
+  const bindMenuProps = bindMenu(popupState);
+
   return (
     <Menu
-      {...bindMenu(popupState)}
+      {...bindMenuProps}
+      onClose={() => {
+        // run custom handler before handler of `bindMenu`
+        onClose();
+        return bindMenuProps.onClose();
+      }}
```

Comment:
When right-clicking with the menu open this happens:

![menu](https://user-images.githubusercontent.com/37453800/193028666-c5ce59dd-5f81-44a8-b9d2-c9ac6db34d3f.gif)


Suggestion fix (probably want to move the `onClose` callback to a function so we can resuse it in both places):

```suggestion
      onClose={() => {
        // run custom handler before handler of `bindMenu`
        onClose();
        return bindMenuProps.onClose();
      }}
       onContextMenu={(event) => {
       event.stopPropagation();
       event.preventDefault();
        // run custom handler before handler of `bindMenu`
        onClose();
        return bindMenuProps.onClose();
      }}
```

Reply:
Wow, what a shame that I missed this, thanks for catching!

Reply:
Fixed here https://github.com/hashintel/hash/pull/1118/commits/ed6994c616027dc544c6d6a85de1df37224280b9!